cirq/ops/eigen_gate.py

```diff
@@ -322,6 +322,32 @@ def _resolve_parameters_(self: TSelf, param_resolver) -> TSelf:
         return self._with_exponent(
                 exponent=param_resolver.value_of(self._exponent))
 
+    def _equal_up_to_global_phase_(self, other, atol):
+        if not isinstance(other, EigenGate):
+            return NotImplemented
+
+        exponents = (self.exponent, other.exponent)
+        exponents_is_parameterized = tuple(
+            protocols.is_parameterized(e) for e in exponents)
+        if (all(exponents_is_parameterized) and exponents[0] != exponents[1]):
```

Comment:
What is the reason that you needed these checks for? I would have expected that copying with global shift set to 0 and then delegating to approx_eq would have covered the parameterized case.

Reply:
This check addressed parameterized exponents only, whereas `approx_eq` is passed the gates stripped of their exponents.

At one point, I was trying to get it so that gates would be equal up to global phase if their exponents were off by half a period (so a phase of `-1`), but it got too complicated to be worth it. As it is, the checking of exponents modulo the period might be better put in `_approx_eq_`.